Utils/ExtractVersionFromTilsesetJSON/compute_versiondata_from_py3dfiles.py

```diff
@@ -10,182 +9,212 @@
-  data = json.load(json_file)
-  VersionTransition = warlock.model_factory(data)
-
-v1 = Version(id="v1",
-             name="2009",
-             description="Limonest state in 2009 for the concurrent point of view",
-             startDate="2009",
-             endDate="2010",
-             tags=["concurrent"],
-             featuresIds=list(get_featuresid(transactions, 2009)['version']))
-
-v2 = Version(id="v2",
-             name="2012",
-             description="Limonest state in 2012 for the concurrent point of view",
-             startDate="2012",
-             endDate="2013",
-             tags=["concurrent"],
-             featuresIds=list(get_featuresid(transactions, 2012)['version']))
-
-v3 = Version(id="v3",
-             name="2015",
-             description="Limonest state in 2015 for the concurrent point of view",
-             startDate="2015",
-             endDate="2016",
-             tags=["concurrent"],
-             featuresIds=list(get_featuresid(transactions, 2015)['version']))
-
-vt_v1_v2 = VersionTransition({"id":"vt1",
-                             "name":"v1->v2",
-                             "startDate":"2009",
-                             "endDate":"2012",
-                             "from":"v1",
-                             "to":"v2",
-                             "description":"Transition between v1 and v2",
-                             "type":"realized",
-                             "transactionsIds":list(get_featuresid(transactions, 2009)['versionTr'])})
-
-vt_v2_v3 = VersionTransition({"id":"vt2",
-                             "name":"v2->v3",
-                             "startDate":"2012",
-                             "endDate":"2015",
-                             "from":"v2",
-                             "to":"v3",
-                             "description":"Transition between v2 and v3",
-                             "type":"realized",
-                             "transactionsIds":list(get_featuresid(transactions, 2012)['versionTr'])})
-
-with open('./data/tileset.json') as json_file:
-  data = json.load(json_file)
-
-data['extensions']['3DTILES_temporal']['versions'] = [v1, v2, v3]
-data['extensions']['3DTILES_temporal']['versionTransitions'] = [vt_v1_v2, vt_v2_v3]
-
-with open('./data/new_tileset.json', "w") as json_file:
-  json.dump(data, json_file)
-
-log('done')
-
-"""# TO SOLVE
-
-* Should version has a start and end date or only a moment of existence ?
-* What to do with aggregated transition ? for the moment (pop)
+    ret = {'version':set(), 'versionTr':set()} # set for batiment's id and transactions's id to have unicity
+    for index, row in df.iterrows():
+        ret['versionTr'].add(row['id'])
+        for s in row[row_name]:
+            ret['version'].add(s) #take the source
+    
+        if row['transactions']==row['transactions']: # separate simple and aggregate transactions (simple has NaN value in the transactions attribut)
```

Comment:
I do it like this because when iterrating over the row. The object is no more a Dataframe. When i check row['transactions']==row['transactions'], it means "if row['transactions'] exists"

Reply:
Ok; can you test if it exists then ? e.g. with `if hasattr(row, 'transactions'):`

Reply:
Sorry my mistake. It doesn't mean that it exists. It means that its value is NaN.